crates/trigger-http/src/wagi.rs

```diff
@@ -73,13 +75,16 @@ impl HttpExecutor for WagiHttpExecutor {
 
         let stdout = MemoryOutputPipe::new(usize::MAX);
 
-        let wasi_builder = instance_builder.factor_builders().wasi();
-
-        // Set up Wagi environment
-        wasi_builder.args(argv.split(' '));
-        wasi_builder.env(headers);
-        wasi_builder.stdin_pipe(Cursor::new(body));
-        wasi_builder.stdout(stdout.clone());
+        if let Some(wasi_builder) = instance_builder
+            .factor_builders()
+            .for_factor::<WasiFactor>()
```

Comment:
> Similarly, we should bail if there is a WASI trigger and no OutboundHttpFactor; unfortunate hard dependency there...

@lann It's unclear to me how a WASI HTTP trigger configured with no `OutboundHttpFactor` would be a no go. Can you explain?

Reply:
`wasmtime-wasi-http` wants to manage inbound and outbound stuff in the same implementation, leading to [this](https://github.com/fermyon/spin/blob/main/crates/factor-outbound-http/src/wasi.rs#L49) and [this](https://github.com/fermyon/spin/blob/09de9c38f6c9ca280020cd9afd5eb655651b8766/crates/trigger-http/src/wasi.rs#L54).

Reply:
That's a shame...

Reply:
It is! It could probably be reworked to have a separate `WasiHttpFactor` that both depend on but...seemed like a lot of work to avoid something that - at least at the time - was just annoying.